### <a name="different-exception-handling-for-objectcontextcreatedatabase-and-dbproviderservicescreatedatabase-methods"></a><span data-ttu-id="f91ea-101">Zpracování pro metody ObjectContext.CreateDatabase a DbProviderServices.CreateDatabase různých výjimek</span><span class="sxs-lookup"><span data-stu-id="f91ea-101">Different exception handling for ObjectContext.CreateDatabase and DbProviderServices.CreateDatabase methods</span></span>

|   |   |
|---|---|
|<span data-ttu-id="f91ea-102">Podrobnosti</span><span class="sxs-lookup"><span data-stu-id="f91ea-102">Details</span></span>|<span data-ttu-id="f91ea-103">Od verze rozhraní .NET Framework 4.5, v případě selhání vytvoření databáze <code>CreateDatabase</code> metody se pokusí o odpojení prázdná databáze.</span><span class="sxs-lookup"><span data-stu-id="f91ea-103">Beginning in .NET Framework 4.5, if database creation fails, <code>CreateDatabase</code> methods will attempt to drop the empty database.</span></span> <span data-ttu-id="f91ea-104">Pokud tuto operace úspěšná, původní <xref:System.Data.SqlClient.SqlException?displayProperty=name> se rozšíří (místo <xref:System.InvalidOperationException?displayProperty=name> , vždy byla vydána v rozhraní .NET Framework 4.0)</span><span class="sxs-lookup"><span data-stu-id="f91ea-104">If that operation succeeds, the original <xref:System.Data.SqlClient.SqlException?displayProperty=name> will be propagated (instead of the <xref:System.InvalidOperationException?displayProperty=name> that was always thrown in .NET Framework 4.0)</span></span>|
|<span data-ttu-id="f91ea-105">Návrh</span><span class="sxs-lookup"><span data-stu-id="f91ea-105">Suggestion</span></span>|<span data-ttu-id="f91ea-106">Při zachytávání <xref:System.InvalidOperationException?displayProperty=name> při provádění <xref:System.Data.Objects.ObjectContext.CreateDatabase> nebo <xref:System.Data.Common.DbProviderServices.CreateDatabase(System.Data.Common.DbConnection,System.Nullable{System.Int32},System.Data.Metadata.Edm.StoreItemCollection)>, SQLExceptions by měl nyní také být zachycena.</span><span class="sxs-lookup"><span data-stu-id="f91ea-106">When catching an <xref:System.InvalidOperationException?displayProperty=name> while executing <xref:System.Data.Objects.ObjectContext.CreateDatabase> or <xref:System.Data.Common.DbProviderServices.CreateDatabase(System.Data.Common.DbConnection,System.Nullable{System.Int32},System.Data.Metadata.Edm.StoreItemCollection)>, SQLExceptions should now also be caught.</span></span>|
|<span data-ttu-id="f91ea-107">Rozsah</span><span class="sxs-lookup"><span data-stu-id="f91ea-107">Scope</span></span>|<span data-ttu-id="f91ea-108">Vedlejší</span><span class="sxs-lookup"><span data-stu-id="f91ea-108">Minor</span></span>|
|<span data-ttu-id="f91ea-109">Version</span><span class="sxs-lookup"><span data-stu-id="f91ea-109">Version</span></span>|<span data-ttu-id="f91ea-110">4.5</span><span class="sxs-lookup"><span data-stu-id="f91ea-110">4.5</span></span>|
|<span data-ttu-id="f91ea-111">Typ</span><span class="sxs-lookup"><span data-stu-id="f91ea-111">Type</span></span>|<span data-ttu-id="f91ea-112">Modul runtime</span><span class="sxs-lookup"><span data-stu-id="f91ea-112">Runtime</span></span>|
|<span data-ttu-id="f91ea-113">Ovlivněné rozhraní API</span><span class="sxs-lookup"><span data-stu-id="f91ea-113">Affected APIs</span></span>|<ul><li><xref:System.Data.Objects.ObjectContext.CreateDatabase?displayProperty=nameWithType></li><li><xref:System.Data.Common.DbProviderServices.CreateDatabase(System.Data.Common.DbConnection,System.Nullable{System.Int32},System.Data.Metadata.Edm.StoreItemCollection)?displayProperty=nameWithType></li></ul>|

